src/OpenLoco/src/Ui/WindowManager.cpp

```diff
@@ -1466,6 +1467,103 @@ namespace OpenLoco::Ui::WindowManager
         invalidateWidget(window->type, window->number, widgetIndex);
     }
 
+    // Allow mouse wheel scrolling to manipulate stepper widgets and tool sizes
+    static bool stepperWheelInput(Window& w, WidgetIndex_t widgetIndex, int32_t wheel)
+    {
+        auto widgetType = w.widgets[widgetIndex].type;
+
+        // If we aren't scrolling on a tool preview or stepper textbox, lower the widget index
+        // once or twice to try to find the parent widget. This allows scrolling on the
+        // increase/decrease buttons themselves, too.
+        int32_t attempts = 0;
+        while (widgetType != WidgetType::wt_3 && widgetType != WidgetType::textbox && widgetIndex > 0)
+        {
+            switch (widgetType)
+            {
+                case WidgetType::toolbarTab: // + and - for tool widget
+                case WidgetType::button:     // + and - for stepper widget
+                {
+                    if (attempts > 0)
+                    {
+                        // Verify that the previous button was of the same type
+                        auto previousType = w.widgets[widgetIndex + 1].type;
```

Comment:
this is the next type not the previous though?

Reply:
It's after the first attempt, so we're comparing this widget to the previous one, right?

Reply:
oh how confusing... Perhaps we should just unroll this.

Reply:
I suppose we could refactor this loop into a lambda that we then run a few times, but I can't quite think of an elegant way of doing so. Do you have a concrete suggestion, or should we just leave it as is?